packages/graphql-dynamodb-transformer/src/DynamoDBModelTransformer.ts

```diff
@@ -137,9 +138,13 @@ export class DynamoDBModelTransformer extends Transformer {
             this.resources.makeIAMRole(typeName)
         )
         ctx.setResource(
-            ModelResourceIDs.ModelTableDataSourceID(typeName),
+            dataSourceRoleLogicalID,
```

Comment:
Wouldn't you need to add this to the stack mapping?

Reply:
https://github.com/aws-amplify/amplify-cli/blob/9ffb02bfa36ec4efbb01d6f093c6e9e811827eb8/packages/graphql-dynamodb-transformer/src/DynamoDBModelTransformer.ts#L102
Already exists a stack mapping that covers this case.